lib/src/settings.rs

```diff
@@ -54,8 +70,16 @@ impl UserSettings {
             config_builder =
                 config_builder.add_source(config::File::from_str(s, config::FileFormat::Toml));
         }
-        self.config = config_builder.build()?;
-        self.timestamp = get_timestamp_config(&self.config, "user.timestamp");
+        let new_config = config_builder.build()?;
+        let new_rng_seed = get_rng_seed_config(&new_config);
+        if new_rng_seed != get_rng_seed_config(&self.config) {
+            // Allow changing the RNG seed, but avoid resetting the RNG when unrelated
+            // options are changed. Care still needs to be taken not to reset the RNG to
+            // another value and then back to the initial value.
+            self.rng.reset(new_rng_seed);
+        }
```

Comment:
Do we need to test if seed value has been changed?

If it's okay to reset RNG by `settings.incorporate_toml_settings(toml_strs_containing_seed)`, I think it should be just fine to reset RNG on every `settings.incorporate_toml_settings()` call.

Reply:
I might be overcautious, but I don't agree. For example, if we add some way for the user to change the config while `jj` is executing (e.g. in a GUI/TUI), they'll change some unrelated config and the RNG will be reset.

Reply:
I'll trust your opinion on whether this is me battling giants (real problems, that is) or windmills.

Reply:
Also, maybe this is too much trouble to go through just to let people change the seed using command-line options.

Reply:
> if we add some way for the user to change the config while jj is executing

In that case, I expect RNG will persist no matter if the seed setting is changed (or TUI/GUI wouldn't provide a way to change the seed.) And I would add another UserSettings method that updates config without resetting RNG.

I mean, it's not important whether toml_strs contains the seed or not, but the phase whether jj is initializing or ready to operate. Does that make sense?

Reply:
Since the random seed should only be set by tests, I'm personally fine with not worrying about the seed changing while the process is running.